File: week_1/copy_file.cpp
#include <iostream>
#include <filesystem>

int main(int argc, char** argv) {
    if (argc != 3) {
        std::cerr << "Usage: " << argv[0] << " <source> <destination>\n";
        return 2;
    }

    try {
        std::filesystem::copy_file(
            argv[1],
            argv[2],
            std::filesystem::copy_options::overwrite_existing
        );
    } catch (const std::exception& e) {
        std::cerr << "Copy failed: " << e.what() << '\n';
        return 1;
    }
    return 0;
}

File: week_1/copy_file.rs
use std::{env, fs, process};

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 3 {
        eprintln!("Usage: {} <source> <destination>", args[0]);
        process::exit(2);
    }
    if let Err(e) = fs::copy(&args[1], &args[2]) {
        eprintln!("Copy failed: {e}");
        process::exit(1);
    }
}
